frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -1310,87 +1311,64 @@ class ResolutionResultByPostorderID {
   /** prepare to resolve the body of a For loop */
   void setupForParamLoop(const uast::For* loop, ResolutionResultByPostorderID& parent);
 
-  ResolvedExpression& byIdExpanding(const ID& id) {
-    auto postorder = id.postOrderId();
-    CHPL_ASSERT(id.symbolPath() == symbolId.symbolPath());
-    CHPL_ASSERT(0 <= postorder);
-    if ((size_t) postorder < vec.size()) {
-      // OK
-    } else {
-      vec.resize(postorder+1);
-    }
-    return vec[postorder];
-  }
-  ResolvedExpression& byAstExpanding(const uast::AstNode* ast) {
-    return byIdExpanding(ast->id());
-  }
-
+  /* ID query functions */
   bool hasId(const ID& id) const {
     auto postorder = id.postOrderId();
     if (id.symbolPath() == symbolId.symbolPath() &&
-        0 <= postorder && (size_t) postorder < vec.size())
+        0 <= postorder && map.count(postorder))
       return true;
 
     return false;
   }
-  bool hasAst(const uast::AstNode* ast) const {
-    return ast != nullptr && hasId(ast->id());
-  }
-
   ResolvedExpression& byId(const ID& id) {
-    CHPL_ASSERT(hasId(id));
     auto postorder = id.postOrderId();
-    return vec[postorder];
+    return map[postorder];
   }
   const ResolvedExpression& byId(const ID& id) const {
     CHPL_ASSERT(hasId(id));
     auto postorder = id.postOrderId();
-    return vec[postorder];
-  }
-  ResolvedExpression& byAst(const uast::AstNode* ast) {
-    return byId(ast->id());
-  }
-  const ResolvedExpression& byAst(const uast::AstNode* ast) const {
-    return byId(ast->id());
-  }
-  ResolvedExpression* byIdOrNull(const ID& id) {
-    if (hasId(id)) {
-      auto postorder = id.postOrderId();
-      return &vec[postorder];
-    }
-    return nullptr;
+    return map.at(postorder);
   }
   const ResolvedExpression* byIdOrNull(const ID& id) const {
     if (hasId(id)) {
       auto postorder = id.postOrderId();
-      return &vec[postorder];
+      return &map.at(postorder);
     }
```

Comment:
It could be faster to do this with `find` (you'd save one lookup / count).

```C++
auto found = map.find(id.postOrderId());
if (found != map.end()) {
  return &found->second;
}
return nullptr;
```

Reply:
I tried this out and found it breaks ~15 tests. Looked into it with the debugger and realized it is because we are now missing the `id.symbolPath() == symbolId.symbolPath()` check from `hasId`. Apparently there are some instances in `convert-uast` where we call `byIdOrNull` with an `ID` with a different symbol path than our RRBPID, but with the same postorder id, expecting a result of null. That seems ok to me, so I will return this code to how it was previously.